packages/hash/frontend/src/pages/_app.page.tsx

```diff
@@ -131,18 +128,112 @@ const App: FunctionComponent<AppProps> = ({
         };
       `}
       />
-    </ApolloProvider>
+    </>
   );
 };
 
-const AppWithTypeSystemContextProvider: FunctionComponent<AppProps> = (
+// Based on the definition of `NextPage`
+export type AppPage<P = {}, IP = P> = NextComponentType<AppContext, IP, P>;
+
+const AppWithTypeSystemContextProvider: AppPage<AppProps, AppInitialProps> = (
   props,
-) => (
-  <TypeSystemContextProvider>
-    <SessionProvider>
-      <App {...props} />
-    </SessionProvider>
-  </TypeSystemContextProvider>
-);
+) => {
+  const { initialAuthenticatedUser } = props;
+
+  return (
+    <TypeSystemContextProvider>
+      <ApolloProvider client={apolloClient}>
+        <AuthInfoProvider initialAuthenticatedUser={initialAuthenticatedUser}>
+          <App {...props} />
+        </AuthInfoProvider>
+      </ApolloProvider>
+    </TypeSystemContextProvider>
+  );
+};
+
+const redirectInGetInitialProps = (params: {
+  appContext: AppContext;
+  location: string;
+}) => {
+  const {
+    appContext: {
+      ctx: { req, res },
+      router,
+    },
+    location,
+  } = params;
+
+  if (req && res) {
+    if (!res.writeHead) {
+      // `res.writeHead` is not a function whilst building next, so return instead.
+      return;
+    }
+    res.writeHead(307, { Location: location });
+    res.end();
+  } else {
+    void router.push(location);
+  }
+};
```

Comment:
Same, can this be in a separate folder to keep this file easier to read?

Reply:
I've moved this function in to a separate file as part of https://github.com/hashintel/hash/pull/1637/commits/22b40db8e145c4776c7a95c1a1099ed6010bc4f6